internal/integration/entity_synthesis.go

```diff
@@ -0,0 +1,172 @@
+// Package integration ...
+// Copyright 2021 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package integration
+
+import (
+	"sort"
+	"strings"
+
+	sdk_metadata "github.com/newrelic/infra-integrations-sdk/v4/data/metadata"
+	"github.com/newrelic/nri-prometheus/internal/pkg/labels"
+)
+
+// Synthesizer group of rules to synthesis entities
+type Synthesizer struct {
+	rulesByConditions map[Condition]EntityRule
+}
+
+// NewSynthesizer initialize and return a Synthesizer from a set of EntityRules
+func NewSynthesizer(entitySynthesisDefinitions []SynthesisDefinition) Synthesizer {
+	s := Synthesizer{
+		rulesByConditions: make(map[Condition]EntityRule),
+	}
+	for _, ed := range entitySynthesisDefinitions {
+		ed.tagRules = ed.Tags.getTagRules()
+		s.addConditions(ed.EntityRule)
+
+		for _, er := range ed.Rules {
+			er.EntityType = ed.EntityType
+			er.tagRules = er.Tags.getTagRules()
+			er.tagRules = append(er.tagRules, ed.EntityRule.tagRules...)
+			s.addConditions(er)
+		}
+	}
+	return s
+}
+func (s *Synthesizer) addConditions(rule EntityRule) {
+	if rule.Identifier == "" || rule.Name == "" || rule.EntityType == "" {
```

Comment:
Just  thinking if defining a method in EntityRule would make easy to understand this  validation

```go
func (rule *EntityRule) isValid() bool{
	return  rule.Identifier != "" && rule.Name != "" && rule.EntityType != "" 
}
```
Don't know if ìsVali would be the correct name for the function or maybe other name suits better but what I mean is this piece of code only belongs to the EntityRule and the type should provide a method for check it

```go
unc (s *Synthesizer) addConditions(rule EntityRule) {
	if !rule.isValid() {
...
}
```

Reply:
yes now is more readable , thanks